pkg/securityscan/jobHandler.go

```diff
@@ -187,8 +197,40 @@ func (c *Controller) createClusterScanReport(outputBytes []byte, scan *v1.Cluste
 
 func (c *Controller) ensureCleanup(scan *v1.ClusterScan) error {
 	var err error
-	configmaps := c.coreFactory.Core().V1().ConfigMap()
+	// Delete the dameonset
+	daemonsets := c.appsFactory.Apps().V1().DaemonSet()
+	dsPrefix := "sonobuoy-rancher-kube-bench-daemon-set"
+	dsList, err := daemonsets.Cache().List(v1.ClusterScanNS, labels.Set(SonobuoyWorkerLabel).AsSelector())
+	if err != nil {
+		return fmt.Errorf("cis: ensureCleanup: error listing daemonsets: %v", err)
+	}
+	for _, ds := range dsList {
+		if !strings.HasPrefix(ds.Name, dsPrefix) {
+			continue
+		}
+		if e := daemonsets.Delete(v1.ClusterScanNS, ds.Name, &metav1.DeleteOptions{}); e != nil && !errors.IsNotFound(e) {
+			return fmt.Errorf("cis: ensureCleanup: error deleting daemonset %v: %v", ds.Name, e)
+		}
+	}
+
+	// Delete the pod
+	pods := c.coreFactory.Core().V1().Pod()
```

Comment:
Can this be initialized on the controller?

Reply:
done